Core/Src/bms.c

```diff
@@ -43,21 +43,20 @@ void bms_init()
 		osTimerNew(&bms_fault_callback, osTimerOnce, NULL, NULL);
 
 	bms_monitor_queue =
-		osMessageQueueNew(CAN_QUEUE_SIZE, sizeof(can_msg_t), NULL);
+		osMessageQueueNew(BMS_QUEUE_SIZE, sizeof(can_msg_t), NULL);
 	assert(bms_monitor_queue);
 }
 
-void vBMSCANMonitor(void *pv_params)
+void vAMSCANMonitor(void *pv_params)
 {
 	can_msg_t msg_from_queue;
 
 	osTimerStart(bms->bms_monitor_timer, BMS_CAN_MONITOR_DELAY);
 	for (;;) {
-		if (osOK == osMessageQueueGet(bms_monitor_queue,
-					      &msg_from_queue, NULL,
-					      osWaitForever)) {
-			osTimerStart(bms->bms_monitor_timer,
-				     BMS_CAN_MONITOR_DELAY);
-		}
+		osThreadFlagsWait(NEW_AMS_MSG_FLAG, osFlagsWaitAny,
+				  osWaitForever);
+		osMessageQueueGet(bms_monitor_queue, &msg_from_queue, NULL,
+				  osWaitForever);
```

Comment:
All osWaitForever should prob be 0 so it uses try semantics and therefore will never result in a check for a context switch.

Reply:
Yes there should also be a while here